lib/charms/mongodb/v1/mongos.py

```diff
@@ -61,7 +61,7 @@ def uri(self):
         # Auth DB should be specified while user connects to application DB.
         auth_source = ""
         if self.database != "admin":
-            auth_source = "&authSource=admin"
+            auth_source = "authSource=admin"
```

Comment:
why the "&" is removed?

Reply:
when `admin` isn't the database this `&` causes immediate failure due to PyMongo URI parsing and the charm becomes unrecoverable.

Reply:
got it, thank you for the explanation